docs/architecture/clients/dependency-injection.md

```diff
@@ -0,0 +1,219 @@
+exporting services through feature modules, it can be tricky to manage their lifetime, and it's easy
+to end up with instances of the service being created.
+
+```ts
+@Injectable({
+  provideIn: "root",
+})
+export class MyService {}
+```
+
+### Angular components
+
+Services that do are not stateful and/or are only supposed to be used locally to extract complex
+logic from a component can be injected directly into standalone components, completely bypassing
+modules.
+
+```ts
+@Component({
+  providers: [
+    // This example assumes that MyService uses @Injectable decorators and does not have an abstract interface
+    safeProvider(MyService),
+  ],
+  standalone: true,
+})
+export class MyComponent {}
+```
+
+### Non-Angular contexts
+
+Non-Angular contexts do not use a DI framework and must manually instantiate their dependencies **in
+the correct order**. Instantiating dependencies out of order may result in null values being
+injected into services, causing runtime errors.
+
+## DI by client
+
+### Shared Angular DI
+
+`libs/angular` contains `jslib-services.module.ts`, a services module which registers common
+dependencies used across Angular Typescript clients (web, browser and desktop).
+
+A client may override this DI if a client-specific configuration is required.
+
+### Web
+
+`core.module.ts` contains the main services module for the web client. It imports
+`JslibServicesModule`.
+
+However, the web module heavily uses [feature modules](https://angular.io/guide/feature-modules) to
+divide its code by feature. Feature modules should configure their own DI wherever possible for
+feature-specific (non-global) services.
+
+### Browser
+
+Browser DI is split across the following locations:
+
+- `services.module.ts` is the main services module for the browser extension popup. It imports
+  `JslibServicesModule`
+- `main.background.ts` manually instantiates services used in the background page (for manifest v2)
+  or the service worker (for manifest v3). It does not use any dependency injection framework
+- a series of exported factory functions with the naming convention `[name]-service.factory.ts`.
+  These may be used in manifest v3 service workers in the future
+
+The background page/service worker still does a lot of work in the browser extension, so many
+services are registered in all the above locations.
+
+### Desktop
+
+Desktop DI is split across the following locations:
+
+- `services.module.ts` is the main services module for the Electron renderer process. It imports
+  `JslibServicesModule`
+- `main.ts` manually instantiates services used in the main process. It does not use any dependency
+  injection framework
+
+The main process is primarily used for Electron behavior and native integrations (e.g. storage,
+biometrics, window behavior), so it registers a smaller subset of services. Many services are only
+registered in the renderer process.
```

Comment:
suggestion: maybe clarify that the `renderer process` is basically the "web page" part of Electron (that is how it is, right..? 😅 )

Reply:
It sure is! done